test/redis/interval_cache/setter_test.exs

```diff
@@ -0,0 +1,52 @@
+defmodule Castle.RedisIntervalCacheSetterTest do
+  use Castle.RedisCase, async: false
+  use Castle.TimeHelpers
+
+  @moduletag :redis
+
+  import Mock
+  alias Castle.Redis.Conn, as: Conn
+  alias Castle.Redis.Interval.Keys, as: Keys
+  alias Castle.Redis.Interval.Setter, as: Setter
+
+  @prefix "interval.cache.setter.test"
+
+  setup do
+    redis_clear("#{@prefix}*")
+    from = get_dtim("2017-03-22T01:15:00Z")
+    to = get_dtim("2017-03-22T02:15:00Z")
+    [from: from, to: to]
+  end
+
+  test "sets ttl according to the time" do
+    with_mock Timex, [:passthrough], [now: fn() -> get_dtim("2017-03-22T01:15:00Z") end] do
+      assert Setter.interval_ttl(get_dtim("2017-03-22T00:15:00Z")) == 0
+      assert Setter.interval_ttl(get_dtim("2017-03-22T00:16:00Z")) == 300
+      assert Setter.interval_ttl(get_dtim("2017-03-22T01:14:00Z")) == 300
+      assert Setter.interval_ttl(get_dtim("2017-03-22T01:15:00Z")) == 300
+      assert Setter.interval_ttl(get_dtim("2017-03-22T02:20:00Z")) == 300
+    end
+  end
+
+  test "sets empty counts", %{from: from, to: to} do
+    assert redis_count("#{@prefix}*") == 0
+    Setter.set(@prefix, from, to, %{})
+    assert redis_count("#{@prefix}*") == 1
+    assert Conn.command(["TTL", List.first(redis_keys("#{@prefix}*"))]) == -1
```

Comment:
so this last line is checking that the setter didn't put a TTL on the empty count, right?

Reply:
It's actually checking that it didn't set a TTL, because `to` is more than `@current_interval_buffer` (1 hour) in the past.  And after an hour, we can assume that kinesis/bigquery have processed any impressions/downloads for that interval, and just cache the result from BQ forever.